EpamWebSiteTestingFramework/Epam.TestAutomation.Pages/PageObjects/Pages/JobListingsPage.cs

```diff
@@ -0,0 +1,111 @@
+
+    public Label SkillsSection => new Label(By.XPath("//*[@class='default-label'][contains(text(), 'All Skills')]"));
+
+    public Dropdown SkillsDropdownPanel =>
+        new Dropdown(By.XPath("//*[@class='multi-select-dropdown' and @aria-hidden='false']"));
+
+    public Checkbox SkillOption(string skill) =>
+        new Checkbox(By.XPath($"//*[@class='checkbox-custom-label'][contains(text(), '{skill}')]"));
+
+    public Label ChosenSkillFilter => new Label(By.XPath("//*[@class='filter-tag']"));
+
+    public Panel FiltersPanel => new Panel(By.Id("jobSearchFilterForm"));
+
+    public JobListingsPage()
+    {
+        _resultPage = new JobSearchResultPage();
+    }
+
+    public override bool IsOpened()
+    {
+        return Browser.Driver.GetUrl().Equals(UiTestSettings.JobListingUrl);
+    }
+
+    public override void OpenUrl()
+    {
+        Browser.Driver.GotToWebPageUrl(UiTestSettings.JobListingUrl);
+        Waiter.WaitForCondition(() => KeywordInputField.IsElementDisplayedOnPage());
+    }
+
+    public JobSearchResultPage SearchJobsByKeyword(string searchWord)
+    {
+        KeywordInputField.SendKeys(searchWord);
+        FindButton.Click();
+        Waiter.WaitForCondition(_resultPage.SearchResultHeadingTitle.IsElementDisplayedOnPage);
+        return new JobSearchResultPage();
+    }
+
+    public JobSearchResultPage SearchJobLocationsByKeyWord(string searchWord)
+    {
+        LocationsDropdown.Click();
+        Waiter.WaitForCondition(DropdownLocationsPanel.IsElementDisplayedOnPage);
+        LocationInput.SendKeys(searchWord);
+        ChosenCityLine.Click();
+        Waiter.WaitForCondition(SelectedLocationChosen(searchWord).IsElementDisplayedOnPage);
+        return new JobSearchResultPage();
+    }
+
+    public JobSearchResultPage SearchJobSkillsByKeyWord(string searchWord)
+    {
+        SkillsSection.Click();
+        Waiter.WaitForCondition(() => !SkillsDropdownPanel.GetTextFromAttribute("class").Contains("hidden"));
+        SkillOption(searchWord).Click();
+        FindButton.Click();
+        Waiter.WaitForCondition(ChosenSkillFilter.IsElementDisplayedOnPage);
+        return new JobSearchResultPage();
+    }
+
+    public JobSearchResultPage FillFiltersWithSearchJobData(string job, string skill, string location)
+    {
+        KeywordInputField.SendKeys(job);
+        LocationsDropdown.Click();
+        Waiter.WaitForCondition(DropdownLocationsPanel.IsElementDisplayedOnPage);
+        LocationInput.SendKeys(location);
+        ChosenCityLine.Click();
+        SkillsSection.Click();
+        Waiter.WaitForCondition(() => !SkillsDropdownPanel.GetTextFromAttribute("class").Contains("hidden"));
+        SkillOption(skill).Click();
+        FindButton.Click();
+        return new JobSearchResultPage();
+    }
+
+    public JobSearchResultPage FillFiltersWithSearchJobDataToGetErrorMessage(string job, string location)
```

Comment:
It is better to make a more general method for filling fields. Not always a combination of parameters will return an error. You can use FillFiltersWithSearchJobData method. Just add conditions 
if  (job != null) 
{KeywordInputField.SendKeys(job);
        LocationsDropdown.Click();}
and the same for all parameters